docs/LoaderApplicationInterface.md

```diff
@@ -0,0 +1,663 @@
+
+To enable a layer, or layers, simply pass the name of the layers you wish to
+enable in the `ppEnabledLayerNames` field of the `VkInstanceCreateInfo` during
+a call to `vkCreateInstance`.
+Once done, the layers you have enabled will be active for all Vulkan functions
+using the created `VkInstance`, and any of its child objects.
+
+**NOTE:** Layer ordering is important in several cases since some layers
+interact with each other.
+Be careful when enabling layers as this may be the case.
+See the [Overall Layer Ordering](#overall-layer-ordering) section for more
+information.
+
+The following code section shows how you would go about enabling the
+VK_LAYER_KHRONOS_validation layer.
+
+```
+char *instance_validation_layers[] = {
+    "VK_LAYER_KHRONOS_validation"
+};
+const VkApplicationInfo app = {
+    .sType = VK_STRUCTURE_TYPE_APPLICATION_INFO,
+    .pNext = NULL,
+    .pApplicationName = "TEST_APP",
+    .applicationVersion = 0,
+    .pEngineName = "TEST_ENGINE",
+    .engineVersion = 0,
+    .apiVersion = VK_API_VERSION_1_0,
+};
+VkInstanceCreateInfo inst_info = {
+    .sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO,
+    .pNext = NULL,
+    .pApplicationInfo = &app,
+    .enabledLayerCount = 1,
+    .ppEnabledLayerNames = (const char *const *)instance_validation_layers,
+    .enabledExtensionCount = 0,
+    .ppEnabledExtensionNames = NULL,
+};
+err = vkCreateInstance(&inst_info, NULL, &demo->inst);
+```
+
+At `vkCreateInstance` and `vkCreateDevice`, the loader constructs call chains
+that include the application specified (enabled) layers.
+Order is important in the `ppEnabledLayerNames` array; array element 0 is the
+topmost (closest to the application) layer inserted in the chain and the last
+array element is closest to the driver.
+See the [Overall Layer Ordering](#overall-layer-ordering) section for more
+information on layer ordering.
+
+**NOTE:** *Device Layers Are Now Deprecated*
+> `vkCreateDevice` originally was able to select layers in a similar manner to
+`vkCreateInstance`.
+> This lead to the concept of "instance layers" and "device layers".
+> It was decided by Khronos to deprecate the "device layer" functionality and
+> only consider "instance layers".
+> Therefore, `vkCreateDevice` will use the layers specified at
+`vkCreateInstance`.
+> Because of this, the following items have been deprecated:
+> * `VkDeviceCreateInfo` fields:
+>   * `ppEnabledLayerNames`
+>   * `enabledLayerCount`
+> * The `vkEnumerateDeviceLayerProperties` function
+
+
+### Meta-Layers
+
+Meta-layers are layers which really just contain an ordered list of other
+layers.
+This is to allow grouping layers together in a specified order so that they can
+interact properly.
+Originally, this was designed to group together the individual Vulkan
+Validation layers in the proper order to avoid conflicts.
+While the Vulkan Validation layer no longer uses this, VkConfig does use this
+to group layers together based on user's desires.
```

Comment:
```suggestion
Originally, this was used to group together the individual Vulkan Validation layers in the proper order
to avoid conflicts. It was necessary because instead of a single Validation layer, validation was split into
multiple component layers. The new `VK_LAYER_KHRONOS_validation` layer pulled everything
into a single layer, dropping the need for meta layers. While not necessary for validation anymore,
VkConfig does use meta layers to group layers together based on user's preferences.
```